### PR TITLE
feat(connector): [Airwallex] add multiple redirect support for 3DS

### PR DESCRIPTION
**Pull request by SamraatBansal**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Add multiple redirects to complete the 3DS flow for Airwallex
- The First redirect is done to collect the Device Information where the `attempt_status` is `DeviceDataCollectionPending`
- The Second redirect is required to collect user input like OTP, where the `attempt_status` is `AuthenticationPending`

The acs or verification response that we get from these redirects after user interaction needs to be submitted to Airwallex via [Continue Confirm ](https://www.airwallex.com/docs/api#/Payment_Acceptance/Payment_Intents/_api_v1_pa_payment_intents__id__confirm_continue/post) API Call individually in sequence. 

Airwallex 3DS flow: 

3DS

### Additional Changes

An additional variant (`DeviceDataCollectionPending`) is added to `AttemptStatus` in the DB to accommodate an additional status of the transaction while going through multiple redirects in 3DS flow.

Command : `ALTER TYPE "AttemptStatus" ADD VALUE IF NOT EXISTS 'device_data_collection_pending';`

- [X] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

I ran all the unit tests which shows that this change does not affect any of the previously implemented flows.

Screenshot 2023-03-16 at 4 59 32 PM

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed submitted code
- [X] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by jagan-jaya — DISMISSED** [review]

**Review by jagan-jaya — DISMISSED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -1189,6 +1189,8 @@ pub struct Metadata {
     #[schema(value_type = Object, example = r#"{ "city": "NY", "unit": "245" }"#)]
     #[serde(flatten)]
     pub data: pii::SecretSerdeValue,
+    /// Payload coming in request as a metadata field
+    pub payload: Option<serde_json::Value>,
```
Comment (jarnura) [inline crates/api_models/src/payments.rs:1193]:
```suggestion
    pub payload: Option<pii::SecretSerdeValue>,
```

File: crates/router/src/connector/airwallex/transformers.rs [hunk 2]
```diff
@@ -140,6 +140,39 @@ impl<F, T> TryFrom<types::ResponseRouterData<F, AirwallexAuthUpdateResponse, T,
...
+pub struct AirwallexCompleteRequest {
+    request_id: String,
+    three_ds: AirwallexThreeDsData,
+    #[serde(rename = "type")]
+    three_ds_type: AirwallexThreeDsType,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct AirwallexThreeDsData {
+    acs_response: Option<serde_json::Value>,
```
Comment (jarnura) [inline crates/router/src/connector/airwallex/transformers.rs:153]:
```suggestion
    acs_response: Option<pii::SecretSerdeValue>,
```

File: crates/router/src/core/payments.rs [hunk 3]
```diff
@@ -330,6 +339,47 @@ impl PaymentRedirectFlow for PaymentRedirectCompleteAuthorize {
...
+                let startpay_url = payments_response
+                    .next_action
+                    .and_then(|next_action| next_action.redirect_to_url)
+                    .ok_or(errors::ApiErrorResponse::InternalServerError)
+                    .into_report()
+                    .attach_printable(
+                        "did not receive redirect to url when status is requires customer action",
+                    )?;
+                Ok(api::RedirectionResponse {
+                    return_url: String::new(),
```
Comment (jarnura) [inline crates/router/src/core/payments.rs:364]: @Narayanbhat166 Can we pass return_url as empty string?

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — COMMENTED** [review]
@Abhicodes-crypto For Redirection endpoint changes, in compatibility layer do we need to do a change?

File: crates/router/src/core/payments.rs (hunk 3 above)
Comment (Narayanbhat166) [reply crates/router/src/core/payments.rs:367]: Yes return_url_with_query_params is the one being sent in response

**Review by jagan-jaya — APPROVED** [review]

**Comment by Abhicodes-crypto** [comment]
@jarnura , changes are not needed in compatibility layer.

**Pull request merged**